Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Collections;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
+{
+    [SerializeField]
+    private int width;
+
+    [SerializeField]
+    private int height;
+
+    public Transform Brick;
+    private int[,] Maze;
+    private List<Vector3> pathMazes = new List<Vector3>();
+    private Stack<Vector2> _tiletoTry = new Stack<Vector2>();
```

Comment:
incorrect naming convention. Just name it `tileToTry`. It could also just be a clearer name considering it is a stack